pylint/checkers/base.py

```diff
@@ -1719,7 +1719,31 @@ def _create_naming_options():
     return tuple(name_options)
 
 
-class NameChecker(_BasicChecker):
+class NameCheckerHelper:
```

Comment:
Not sure about the location of this class. Does `pylint/checkers/base_checker.py` perhaps makes more sense? We need a place to define sub-class of `BaseChecker` that get inherited by other classes in other files..

Reply:
Makes sense to use a linear inheritance and to put in withing base_checker.
I will do this.

Reply:
moved